### PR TITLE
Adds productFeatures, DeliveryDate, PriceBaseFix parsing in orderResponse

### PR DESCRIPTION
**Pull request by guiohm**
Also a fix for Country/CountryCode + minor other things

### CODE REVIEW

**Comment by guiohm** [comment]
> Hi,
> 
> good additions! do you have plans to make a pull request to the master repo? That'd be great.
> Concerning 'Country'. How about making it read only and lookup the country name using the Countries class?
> 
> `public string Country { get { return Countries.GetCountry(CountryCode); }`
> 
> ?

Well, this field is actually a part of the schema and can be transmitted without countryCode, so I'm not sure about that.
In the data I need to parse, the supplier use both of them.

File: OpenTrans.net/DeliveryDate.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ using System;
+
+namespace OpenTrans.net
+{
+    public class DeliveryDate
+    {
```
Comment (stephanstapel) [inline OpenTrans.net/DeliveryDate.cs:25]: Could you add the DeliveryDate type as well?

File: OpenTrans.net/PriceBaseFix.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+namespace OpenTrans.net
+{
+    public class PriceBaseFix
+    {
+        public decimal PriceUnitValue { get; set; }
+        public QuantityCodes PriceUnit { get; set; }
```
Comment (stephanstapel) [inline OpenTrans.net/PriceBaseFix.cs:25]: could you add the PriceUnitFactor as well and add documentation (I usually just copy and adapt the text from the specification)?

File: OpenTrans.net/ProductPriceFix.cs [hunk 3]
```diff
@@ -16,15 +16,13 @@
...
 
 namespace OpenTrans.net
 {
     public class ProductPriceFix
     {
-        public decimal PriceAmount {get;set;}
+        public decimal PriceAmount { get; set; }
         public decimal PriceQuantity { get; set; }
+        public PriceBaseFix PriceBaseFix { get; set; }
     }
```
Comment (stephanstapel) [inline OpenTrans.net/ProductPriceFix.cs:27]: could you add the remaining elements here as well?
According to the specification:

**Review by stephanstapel — COMMENTED** [review]
thanks for your answer. Nevertheless my approach would be to be as clean as possible with the data.
If your customer sends both information, what would be the master information? the country code or the country text?

What is your recommendation?

I'd then go ahead and handle the possible conflicts in the reader classes.

**Comment by guiohm** [comment]
About Country / CountryCode, as part of a postal address object, the first is probably more relevant. I see that the later has been added in 2005.
In the Country getter, we could do : return it if it was set, else try to get it from countryCode.
But the issue with that is we loose the ability to know if the Country node is set or not.

The other option is a dedicated method (or extension method) GetCountry() that explains the rule.
Not sure if you want to take this route.

I would prefer having a predictable / transparent library that sticks to the standard (meaning, I can just use the opentrans documentation) and that returns basic objects. Then for those wanting more sugar, extension methods could suit them.

**Comment by guiohm** [comment]
About your other comments, I have this in mind also, but I'm on a tight schedule, so I'll see If I can find a bit of time in the next few days.

**Comment by stephanstapel** [comment]
Implemented most of my comments myself. Please pull the changes into your repo.

**Pull request merged**